src/adminDashboard/adminDashboard.ts

```diff
@@ -51,14 +57,100 @@ export class AdminDashboard {
   async hydrate(): Promise<void> {
     if (this.ethereumService.defaultAccountAddress) {
       const defaultAccount = this.ethereumService.defaultAccountAddress.toLowerCase();
-      this.seeds = this.seedService.seedsArray
-        .filter((seed) => { return seed.admin.toLowerCase() === defaultAccount;});
+      this.seeds = this.seedService.seedsArray;
     } else {
       this.seeds = [];
     }
   }
 
-  connect():void {
+  async setSeed(index) {
+    this.selectedSeed = this.seeds[index];
+  }
+
+  async fundSeed() {
+    try {
+      await this.selectedSeed.projectTokenContract.transfer(this.selectedSeed.address, this.selectedSeed.seedAmountRequired);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async retrieveProjectTokens(receiver: Address) {
+    try {
+      await this.selectedSeed.contract.callStatic.retrieveSeedTokens(receiver);
+      await this.selectedSeed.contract.retrieveSeedTokens(receiver);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async withdrawFundingTokens() {
+    try {
+      await this.selectedSeed.contract.callStatic.withdraw();
+      await this.selectedSeed.contract.withdraw();
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async addWhitelist() {
+    try {
+      const filterPattern = /([^\W]+)/g;
+      const response = await axios.get(this.selectedSeed.metadata.seedDetails.whitelist);
+      const whitelistAddress = response.data.match(filterPattern);
+      await this.selectedSeed.contract.callStatic.whitelistBatch(whitelistAddress);
+      await this.selectedSeed.contract.whitelistBatch(whitelistAddress);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async addToWhitelist(address: Address) {
+    try {
+      await this.selectedSeed.contract.callStatic.whitelist(address);
+      await this.selectedSeed.contract.whitelist(address);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async removeFromWhiteliste(address: Address) {
+    try {
+      await this.selectedSeed.contract.callStatic.unwhitelist(address);
+      await this.selectedSeed.contract.unwhitelist(address);
+    } catch (error) {
+      alert(error.message);
+    }
+  }
+
+  async pauseLaunch() {
+    try {
+      await this.selectedSeed.contract.callStatic.pause();
+      await this.selectedSeed.contract.pause();
```

Comment:
I think it would be better if you add all these Seed functions to the Seed entity.

Reply:
by Seed entity we mean, the SeedService or this.selectedSeed?

Reply:
I have added to this file `src/entities/Seed.ts`.